aip/0191.md

```diff
@@ -81,3 +81,29 @@ these **should** be separated by a blank line:
   corresponding response message (if any).
 - Any remaining `message` definitions.
 - Any top-level `enum` definitions.
+
+### Packaging annotations
+
+Protocol buffers ships with annotations to declare the package or namespace
+(depending on the vocabulary of the target language) of the generated files.
+For example, setting `go_package` or `csharp_namespace` will override the
+inferred package name.
+
+When defining APIs, the following rules apply:
+
+- Java
+  - The `java_package` annotation **must** be set. The correct value is usually
+    the proto package with the appropriate TLD prefixed. Example:
+    `com.google.example.v1`.
+  - The `java_multiple_files` annotation **must** be set to `true`.
+  - The `java_outer_classname` annotation **must** be set, and **should** be
+    set to the name of the proto filename, in `PascalCase`, with `Proto`
+    appended. Example: `LibraryProto`.
+- Other languages
```

Comment:
I was under imporession that at least Go and C# also require those.

Reply:
Both languages have sensible defaults.

Reply:
Can we clarify that? Might it look like we overloooked those languages rather than specifically left them out due to their reasonable defaults?

Reply:
Sure. (I also intend to lint for inconsistency -- e.g. file A has `csharp_namespace` but file B does not.)